CMakeLists.txt

```diff
@@ -1,43 +1,27 @@
+# Set the minimum version of CMake that's required 
 cmake_minimum_required(VERSION 3.12)
 
-project(maier-saupe-lc-hydrodynamics VERSION 0.1 LANGUAGES C CXX CUDA)
+project(maier-saupe-lc-hydrodynamics VERSION 0.1 LANGUAGES C CXX)
 
-# Read in external packages
-find_package(Boost 1.76 COMPONENTS program_options serialization REQUIRED)
-find_package(HighFive REQUIRED)
-find_package(deal.II 9.3.0 REQUIRED)
-deal_ii_initialize_cached_variables()
+# find Deal II
+find_package(deal.II 9.3.0
+  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
+  REQUIRED)
+# massage Deal II into a more modern CMake Target
+if(NOT TARGET deal.II::Interface)
+  add_library(deal.II::Interface INTERFACE IMPORTED GLOBAL)
+  set_target_properties(deal.II::Interface PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${DEAL_II_INCLUDE_DIRS}"
+    INTERFACE_LINK_LIBRARIES "${DEAL_II_LIBRARIES_RELEASE}"
+    )
+endif()
```

Comment:
I've chosen to massage Deal II into a more modern CMake target. This means their documentation about how to use their project with CMake isn't very helpful for us. Basically, their `deal_ii_setup_target` CMake macro does the same thing that modern CMake does when linking except it also auto-detects if the build should be Debug or Release based on how Deal II was built. This feature isn't needed for your end-point project (in my opinion), so it is more helpful to have a modern CMake target that you can link to other targets in the same way as HighFive, Boost::program_options, etc...